File: AkitaModel/BgNhis/ImportRequest.cs
namespace Skyware.Lis.AkitaModel.BgNhis;

/// <summary>
/// Request for NHIS referrals import.
/// </summary>
public class ImportRequest
{

    /// <summary>
    /// Name of requesting application.
    /// </summary>
    public string ApplicationName { get; set; }

    /// <summary>
    /// Processing points location Id.
    /// </summary>
    public string ProcPointLocationId { get; set; }

    /// <summary>
    /// Sale schema.
    /// </summary>
    public Schema Schema { get; set; }

    /// <summary>
    /// Reception location Id.
    /// </summary>
    public string RegLocationId { get; set; }

    /// <summary>
    /// Location Id.
    /// </summary>
    public string LocationId { get; set; }

    /// <summary>
    /// Application user.
    /// </summary>
    public string UserName { get; set; }

    /// <summary>
    /// Id of the existing Sale the referrals must be added.
    /// </summary>
    public int? SaleId { get; set; }

    /// <summary>
    /// Collection of referrals to be imported.
    /// </summary>
    public ReferralImportRequest[] Referrals { get; set; } = [];

}

File: AkitaMotelTS/src/BgNhis/ImportRequest.ts
/**
 * Skyware.Lis.AkitaModel.BgNhis
 */

import { Schema } from './Schema';
import { ReferralImportRequest } from './ReferralImportRequest';

/**
 * Request for NHIS referrals import.
 */
export class ImportRequest {
  /**
   * Name of requesting application.
   */
  applicationName?: string;

  /**
   * Processing points location Id.
   */
  procPointLocationId?: string;

  /**
   * Sale schema.
   */
  schema?: Schema;

  /**
   * Reception location Id.
   */
  regLocationId?: string;

  /**
   * Location Id.
   */
  locationId?: string;

  /**
   * Application user.
   */
  userName?: string;

  /**
   * Id of the existing Sale the referrals must be added.
   */
  saleId?: number;

  /**
   * Collection of referrals to be imported.
   */
  referrals: ReferralImportRequest[] = [];
}
